share/picongpu/tests/FieldAbsorber/lib/python/test/FieldAbsorber/validate.py

```diff
@@ -0,0 +1,176 @@
+    y_offset = -offset_cells
+    z_offset = int(0)
+
+    # Calculate absolute position of measurement point in the small volume.
+    x_meas = Nx//2 + x_offset
+    y_meas = Ny//2 + y_offset
+    z_meas = Nz//2 + z_offset
+
+    # Calculate absolute position of measurement point in the reference sim.
+    x_meas_ref = Nx_ref//2 + x_offset
+    y_meas_ref = Ny_ref//2 + y_offset
+    z_meas_ref = Nz_ref//2 + z_offset
+
+    """Load data"""
+    times = np.zeros(Nt)
+
+    # create data containers.
+    # Explicitly provide data type in order to circumvent
+    # "RuntimeError: [Record_Component::load_chunk()]
+    #  Requires contiguous slab of memory."
+    # of openPMD
+    Ex = np.zeros(Nt, dtype=np.float32)
+    Ey = np.zeros(Nt, dtype=np.float32)
+    Ez = np.zeros(Nt, dtype=np.float32)
+
+    Ex_ref = np.zeros(Nt, dtype=np.float32)
+    Ey_ref = np.zeros(Nt, dtype=np.float32)
+    Ez_ref = np.zeros(Nt, dtype=np.float32)
+
+    i = 0
+    for it in series_test.iterations:
+        times[i] = it
+
+        Efield = series_test.iterations[it].meshes["E"]
+
+        unit_fieldE = Efield["x"].unit_SI
+
+        ##################
+        # Load test data #
+        ##################
+
+        Efield["x"].load_chunk(
+            Ex[i:i+1], offset=(z_meas, y_meas, x_meas), extent=(1, 1, 1)
+        )
+        Efield["y"].load_chunk(
+            Ey[i:i+1], offset=[z_meas, y_meas, x_meas], extent=[1, 1, 1]
+        )
+        Efield["z"].load_chunk(
+            Ez[i:i+1], offset=[z_meas, y_meas, x_meas], extent=[1, 1, 1]
+        )
+
+        # spare the series.flush() since the iteration is closed immediately
+
+        series_test.iterations[it].close()
+
+        Ex[i] *= unit_fieldE
+        Ey[i] *= unit_fieldE
+        Ez[i] *= unit_fieldE
+
+        #######################
+        # Load reference data #
+        #######################
+        Efield_ref = series_ref.iterations[it].meshes["E"]
+
+        Efield_ref["x"].load_chunk(
+            Ex_ref[i:i+1], offset=[z_meas_ref, y_meas_ref, x_meas_ref],
+            extent=[1, 1, 1]
+        )
+        Efield_ref["y"].load_chunk(
+            Ey_ref[i:i+1], offset=[z_meas_ref, y_meas_ref, x_meas_ref],
+            extent=[1, 1, 1]
+        )
+        Efield_ref["z"].load_chunk(
+            Ez_ref[i:i+1], offset=[z_meas_ref, y_meas_ref, x_meas_ref],
+            extent=[1, 1, 1]
+        )
+
+        # spare the series_ref.flush()
+        # since the iteration is closed immediately
+
+        series_ref.iterations[it].close()
+
+        Ex_ref[i] *= unit_fieldE
+        Ey_ref[i] *= unit_fieldE
+        Ez_ref[i] *= unit_fieldE
+
+        i += 1
+
+    del i
+
+    """ Calculate the quantity defining the
+        quality/performance of the absorber.
+    """
+    component, component_ref = (
+        np.sqrt(Ex**2 + Ey**2 + Ez**2),
+        np.sqrt(Ex_ref**2 + Ey_ref**2 + Ez_ref**2)
+    )
+
+    quality = np.abs(component - component_ref)/np.abs(component_ref).max()
```

Comment:
would a component-wise test not also be useful? If components are messed up the amplitude might still appear correct in some cases?

Reply:
In this example, due to the current mainly pointing along y, we effectively test only the y component anyways. The other field components are small compared to it.

To really test all components and boundaries, we would need 6 pairs of small/large simulations where either the current orientation or the wire orientation changes from one to the other.

Reply:
Okay